.github/workflows/ci.yml

```diff
@@ -204,35 +204,41 @@ jobs:
     strategy:
       matrix:
         include:
-          - arch: x86_64
-            image: ubuntu-latest
+          # We want to cover all supported architectures here
+          # FIXME: Add armv7, ppc64, ppc64le
           - arch: aarch64
             image: ubuntu-24.04-arm
+          - arch: loongarch64
+            image: [alpine-edge, loongarch64]
+          - arch: s390x
+            image: [ubuntu-25.10, s390x]
+          - arch: x86_64
+            image: ubuntu-latest
 
     runs-on: ${{ matrix.image }}
 
     steps:
+      # Helpful for debugging and making sure our runners support the
+      # vector instructions we emit
+      - name: Show CPU information
+        run: lscpu
+
       - name: Checkout sources
         uses: actions/checkout@v5
 
-      - name: Install nightly toolchain
+      - name: Install 1.89.0 toolchain
         uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: nightly
-
-      - name: Get latest compatible dependencies for MSRV
-        run: CARGO_RESOLVER_INCOMPATIBLE_RUST_VERSIONS=fallback cargo update
+          toolchain: 1.89.0
```

Comment:
note that `dtolnay/rust-toolchain@1.89` works as well

Reply:
You could also parse the MSRV version from `Cargo.toml` like we do in [Twilight](https://github.com/twilight-rs/twilight/blob/b24a7918ba5e84cbc488a23809b770733b3cefa5/.github/workflows/check.yml#L32C34-L32C88):
```sh
awk '/rust-version/{print $NF}' Cargo.toml | tr -d '"'
```